lib/common/grpc-service.js

```diff
@@ -457,11 +458,12 @@ GrpcService.createDeadline_ = function(timeout) {
 GrpcService.getError_ = function(err) {
   if (err && GRPC_ERROR_CODE_TO_HTTP[err.code]) {
     var defaultErrorDetails = GRPC_ERROR_CODE_TO_HTTP[err.code];
-    err.code = defaultErrorDetails.code;
-    err.message = err.message || defaultErrorDetails.message;
-    return err;
-  }
 
+    return extend(true, new Error(), err, {
```

Comment:
@stephenplusplus do you think we should just use a pojo here instead of a `new Error`?